File: elixir/acronym/acronym.exs
defmodule Acronym do
  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(String.t()) :: String.t()
  def abbreviate(string) do
    # \p{Ll}: a lowercase letter that has an uppercase variant. 
    # \p{Lu}: an uppercase letter that has a lowercase variant
    # \p{P}:  any kind of punctuation character

    Regex.replace(~r/(\p{Ll})(\p{Lu})/, string, "\\1 \\2", global: true)
    |> String.split(~r/[\p{P}\s]/, trim: true)
    |> Enum.map(&(String.slice(&1, 0, 1)))
    |> Enum.join("")
    |> String.upcase
  end
end

File: go/acronym/acronym.go
package acronym

import (
	"regexp"
	"strings"
)

const testVersion = 3

func Abbreviate(l0 string) string {
	l1 := regexp.MustCompile("([\\p{Ll}])([\\p{Lu}])").ReplaceAllString(l0, "$1 $2")

	l2 := regexp.MustCompile("[\\p{P}\\s]").Split(l1, -1)

	l3 := make([]string, 0)

	for i := 0; i < len(l2); i++ {
		runes := []rune(l2[i])
		if len(runes) > 0 {
			l3 = append(l3, string(runes[0]))
		}
	}
	l4 := strings.Join(l3, "")

	l5 := strings.ToUpper(l4)

	return l5
}
